tools/api-key-rotation/api_key_rotation_checker/main.py

```diff
@@ -0,0 +1,348 @@
+            # (API keys are dictionaries)
+            for apikey in apikeys:
+                # Google removed the "createdBy" field
+                # so only legacy keys have it
+                if "createdBy" in apikey:
+                    # Create our API key object
+                    # if it has "createdBy"
+                    key_object = ApiKey(
+                        apikey["keyId"],
+                        apikey["displayName"],
+                        apikey["createdBy"],
+                        apikey["createTime"],
+                        project_id)
+                else:
+                    # Create our API key object
+                    # if it does NOT have "createdBy"
+                    key_object = ApiKey(
+                        apikey["keyId"],
+                        apikey["displayName"],
+                        "None",
+                        apikey["createTime"],
+                        project_id)
+
+                # We need to convert
+                # our creation time for comparison
+                converted_creation_date = time_convert(key_object)
+
+                # Extract API Key ID for logging
+                key_id = key_object.key_id
+
+                # If our key is older than x days (default 90)
+                # based on our compare_dates function
+                # add api key to appropriate variable container
+                logging.info("Checking API key: %s creation date.. \n", key_id)
+                # Convert to JSON for logging
+                key_object_json = key_object.to_json()
+                if compare_dates(converted_creation_date, rotation_date):
+                    keys_needing_rotation.append(key_object_json)
+                else:
+                    keys_not_needing_rotation.append(key_object_json)
+
+    # Format our API keys
+    keys_needing_rotation = "\n".join(keys_needing_rotation)
+    keys_not_needing_rotation = "\n".join(keys_not_needing_rotation)
+
+    # Once analysis is complete for all keys,
+    # log results
+    if keys_needing_rotation:
+        logging.warning(
+            "Found API keys older than %s days. Please rotate: \n%s \n", rotation_period, keys_needing_rotation) # pylint: disable = line-too-long
+
+    if keys_not_needing_rotation:
+        logging.info(
+            "The following API key(s) are not older than %s days: \n%s", rotation_period, keys_not_needing_rotation) # pylint: disable = line-too-long
+
+
+def create_token():
+    """
+    As of March 2021, Google has released a SDK for API keys
+    but the SDK is in a private alpha and not accessible for
+    public usage. You can view the SDK here:
+    https://cloud.google.com/sdk/gcloud/reference/alpha/services/api-keys
+
+    Therefore, we must use the requests library and make API calls
+    with a bearer token until the SDK is usable by the public.
+    """
+    access_token = subprocess.run(
+        "gcloud auth print-access-token",
```

Comment:
```suggestion
        ["gcloud", "auth", "print-access-token"],
```
> Unless otherwise stated, it is recommended to pass args as a sequence.
https://docs.python.org/3/library/subprocess.html#subprocess.Popen